### DISCUSSION TITLE
[EVAL] Has git-lfs been considered for versioning node_modules

### DISCUSSION THREAD

**Discussion by richardeschloss** [npm/rfcs#149] [General]
Hi I just came across [git-lfs](https://git-lfs.github.com/) today, haven't really played with it. I realize I should have done homework before opening the discussion. Just wanted to jot down before I forget.

One of the neat things about git is that it is pretty darn fast when it comes to switching between branches (i.e., changing out file trees). While git can probably be used on node_modules, it's generally not because of the size of dependencies. Perhaps something like git-lfs would be the more appropriate solution for that? Something like that can maybe manage node_modules and a new "node_assets" or "node_artifacts"? 

Then the idea of "installing" deps may shift more to "pulling" or "cloning" deps? Git is pretty good at resolving deltas between branches, so changing project deps might become noticably faster?

**Comment by ljharb** [comment]
imo the reason it's generally not is because generated things ideologically/philosophically shouldn't be in version control, not because of the size.

**Reply by richardeschloss** [reply]
I tend to agree with that and have shared the same views. I can probably have inked my thoughts better. The angle I was getting at was more of lightning-fast installs, installs that would be as fast as pulling git branches, where each dependency *set* would be a snapshot. Let me keep thinking about this, I'm aware my thoughts are not fully baked right now.

**Reply by richardeschloss** [reply]
(i tried editing my original post, but github discussions has the "edit" disabled)

**Reply by ljharb** [reply]
It seems like it'd be pretty awesome to utilize git-lfs for that kind of feature, but i wouldn't want it to show up in my project's repo and be pushed to remotes.

**Reply by richardeschloss** [reply]
Agreed there! I'd only want it for local dev
